test/minikube/001-kube.bats

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bats
+#
+# Tests of podman kube commands with minikube
+#
+
+load ../system/helpers.bash
+
+function setup(){
+    # only set up the minikube cluster before the first test
+    if [ "$BATS_TEST_NUMBER" -eq 1 ]; then
+        minikube start
+        echo $output >&2
+    fi
+    # wait for a bit so all the services in the cluster come up
+    sleep 10
+}
+
+###############################################################################
+# BEGIN tests
+
+@test "minikube - check cluster is up" {
+    run minikube kubectl get nodes
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"Ready"* ]]
+    run minikube kubectl get pods
+    [ "$status" -eq 0 ]
+    [[ "$output" == "No resources found in default namespace." ]]
+}
+
+@test "minikube - deploy generated container yaml to minikube" {
+    cname="test-ctr"
+    fname="/tmp/f$(random_string 6).yaml"
+    run_podman container create --name $cname $IMAGE top
+    [ "$status" -eq 0 ]
+    run_podman kube generate -f $fname $cname
+    [ "$status" -eq 0 ]
+
+    # deploy to the minikube cluster
+    run minikube kubectl -- apply -f $fname
+    [ "$status" -eq 0 ]
+    [[ "$output" == "pod/$cname-pod created" ]]
+    # need to wait for the pod to be created and started before checking status
+    sleep 5
+    run minikube kubectl get pods
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"Running"* ]]
+
+}
+
+@test "minikube - deploy generated pod yaml to minikube" {
+    pname="test-pod"
+    cname1="test-ctr1"
+    cname2="test-ctr2"
```

Comment:
Same random-name + check `exists` suggestion applies here.  Alternatively (for all tests) give podman (and minikube?) a unique root and run-root directory to store state.  Then nobody ever has to worry about clashes (they're super-duper annoying to debug).

Reply:
Where/how can I change the root and run-root directories for these tests?

Reply:
Ugh - that's varsity-level. System tests were written with the specific goal of using the default system setup, so I made no provision for things like that. If you absolutely have to, ac/grep for `_PODMAN_TESTS_OPTS` and `runroot`, and see (e.g.) the 330 and 520 tests.

Reply:
I also don't think this should be a problem because I am calling `basic_teardown()` after each tests which deletes all pods and containers before moving on to the next test.

Reply:
That may be good enough for now.  Especially if there's a failure to remove and it logs that fact.